.github/workflows/block-extractor-scheduler.yml

```diff
@@ -0,0 +1,68 @@
+name: Block Extractor Workflow
+
+on:
+  schedule:
+    - cron: '0 */3 * * *'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  RUST_LOG: info
+
+jobs:
+  backup-blocks:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        database: [postgres, bigquery]
+        network: [mainnet, testnet]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Compile evm-block-extractor
+        run: cargo build --release -p evm-block-extractor --bin evm-block-extractor
+
+      - name: Move binary to root
+        run: mv target/release/evm-block-extractor evm-block-extractor
+
+      - name: Make binary executable
+        run: chmod +x evm-block-extractor
+
+      - name: Set Up Environment Variables
+        run: |
+          if [ "${{ matrix.network }}" = "mainnet" ]; then
+              echo "RPC_URL=mainnet.bitfinity.network" >> $GITHUB_ENV
+              echo "BIGQUERY_DATASET=BIGQUERY_DATASET_${{ matrix.network }}" >> $GITHUB_ENV
```

Comment:
this variable is required only if `matrix.database == 'bigquery'` so it should be in the `Run evm-block-extractor for BigQuery` block